pkg/identityserver/user_registry.go

```diff
@@ -533,6 +571,14 @@ func (is *IdentityServer) updateUser(ctx context.Context, req *ttnpb.UpdateUserR
 		})
 	}
 
+	// If primary email address was updated and invalidated, send a new validation email.
+	if usr.PrimaryEmailAddressValidatedAt == nil &&
```

Comment:
The trigger here should be the contact info change, not the `PrimaryEmailAddressValidatedAt`, as the contact info is what is validated (and ends up also updating `PrimaryEmailAddressValidatedAt`.

Reply:
not sure if I agree with this, the way I see this is that an emailValidateAt will not be `nil` if:
1. An admin is updating it and sets the `PrimaryEmailAddressValidatedAt` in the request body.
2. An admin updates just the `PrimaryEmailAddress` field and expects for it to be validated since the old email was already validated.
3. An admin did the update of `PrimaryEmailAddress` before this fix and therefore its value its not on the `contactInfo` list, after v3.28 a new update of just `PrimaryEmailAddress` field is done making the following true:
  3.1.  no old contact is found
  3.2. there is no `validateAt` for the new email to inherit
  3.3. There won't be a `ContactInfoChange` as true and the new email will be added to the `contactInfo` list
  3.4. The new email is invalidated and has to be validated.

I believe that having the `PrimaryEmailAddressValidatedAt == nil` covers the third case, which is an edge case but it has a good change of happening.

Reply:
It's more of a consistency thing - since `contact_info` is a list, any item of that list could have `validated_at` set to `nil` and would require a contact info validation. I am ok with keeping this as proposed here and getting rid of the mutable contact info ASAP.

Reply:
Left it with pretty much the same logic since the removal of the mutable contact info is the priority after the notification API, so I believe it is okay as is. There is a comment for more clarification around it